test/test-expressroutes.js

```diff
@@ -84,10 +85,11 @@ test('express routes', function (t) {
 
             stack = Array.prototype.slice.call(parent._router.stack, 3);
 
-            t.strictEqual(stack.length, 2, '2 routes added.');
-            t.strictEqual(stack[1].route.path, '/v1/petstore/pets', '/pets added.');
-            t.strictEqual(stack[1].route.stack.length, 2, '/pets has middleware.');
-            t.strictEqual(stack[1].route.stack[0].name, 'm1', '/pets has middleware named m1.');
+            t.strictEqual(stack.length, 3, '3 routes added.');
+            t.strictEqual(stack[1].regexp.toString(), '/^\\/v1\\/petstore\\/pets(?:\\/(?=$))?$/i', 'bleb');
```

Comment:
It's actually because we added a middleware between the 2 exiting ones.
So the previous stack[1] middleware is now in stack[2], and stack[1] now contains the new middleware that doesn't have nor need an attribute 'route', so we just check that the 'regexp' used is all right.
However, the next commit will finally put the new middleware in last (so in stack[2]), so this will all be a little bit clearer.